contracts/strategies/MasterChefStrategy.sol

```diff
@@ -152,9 +157,14 @@ abstract contract MasterChefStrategy is YakStrategyV2 {
     }
 
     function reinvest() external override onlyEOA {
-        (uint256 poolTokenAmount, uint256 rewardTokenBalance, uint256 estimatedTotalReward) = _checkReward();
+        (
+            uint256 poolTokenAmount,
+            uint256 extraTokenAmount,
+            uint256 rewardTokenBalance,
+            uint256 estimatedTotalReward
+        ) = _checkReward();
         require(estimatedTotalReward >= MIN_TOKENS_TO_REINVEST, "MasterChefStrategyV1::reinvest");
-        _reinvest(rewardTokenBalance, poolTokenAmount);
+        _reinvest(rewardTokenBalance, poolTokenAmount, extraTokenAmount);
     }
 
     function _convertPoolTokensIntoReward(uint256 poolTokenAmount) private returns (uint256) {
```

Comment:
Since we're doing already fundamental changes on this, should we check if poolToken == rewardToken then return? We don't want to swap in this condition